### ISSUE TITLE
How to use ToggleNextPane with `Ctrl + Tab`?

### ISSUE THREAD

**Issue by RoneoOrg** [zellij-org/zellij#1962]
Hi there,

I want to navigate between panes with `Ctrl + Tab` .

To do so, I set this in `~/.config/zellij/config.kdl`:

```
keybinds {
    normal {
        bind "Ctrl: Tab" { FocusNextPane; }
```

### Problem

I get the following error when saving the config file and restarting zellij:

```
  $ zellij
  × Failed to parse Zellij configuration
   ╭─[~/.config/zellij/config.kdl:5:1]
      ...
 6 │         bind "Ctrl Tab" { FocusNextPane; }
   ·         ─────────────────┬─────────────────
   ·                          ╰── Invalid key: 'Ctrl Tab'
```

I also tried the following syntaxes: `"Ctrl: Tab"` and `"Ctrl: 'Tab'"`

Note that other keybindings work fine, for example `bind "Tab" { FocusNextPane; }` works perfectly.

It seems to be specific to the `Ctrl` + `Tab` combination. What am I missing?

### Technical details

`zellij --version`: 0.33.0
`stty size`: 55 211
`uname -av`: Linux linux 5.10.0-19-amd64 #1 SMP Debian 5.10.149-2 (2022-10-21) x86_64 GNU/Linux
`gnome-terminal --version`: GNOME Terminal 3.38.3 using VTE 0.62.3 +BIDI +GNUTLS +ICU +SYSTEMD

Any hint is welcome, Thanks !

**Comment by mnpqraven** [comment]
I don't think you can bind Ctrl + Tab as a valid combination. I was using Ctrl Tab with tmux and when I was trying out zellij I realized that I can't set that keybind too.

**Comment by naosense** [comment]
I'm wondering if it's not supported in kdl configuration or not supported in zj

**Comment by naosense** [comment]
After i checked the code base, i found this
https://github.com/zellij-org/zellij/blob/2641cccafccace33e20d408489e1832b75757c28/zellij-utils/src/data.rs#L75-L83

Hey @imsnif , sorry to bother, why the `main_key` must be single character key?

**Comment by imsnif** [comment]
Hey @naosense - sorry for the late reply! I was moving to a new apartment over the weekend and was very busy.

While it is possible to change the `Key` struct to allow for other combinations, the real limitation is in what we can read from STDIN. Things like ctrl+tab *may* be possible, but I'm not 100% sure.

Before changing the struct, I'd recommend doing some manual tests and seeing what sort of input ctrl+tab (or any other key we'd like to check) gives us. You can add a `log::info` line after this line here: https://github.com/zellij-org/zellij/blob/main/zellij-client/src/stdin_handler.rs#L44 and see.

Even if this doesn't work and we don't see a difference between ctrl+tab and eg. just tab, it might be an issue with `termwiz`, our STDIN parsing library. If that's the case, we'll see what (if anything) we can do.

**Comment by naosense** [comment]
Hey @imsnif,  i added log after here
https://github.com/zellij-org/zellij/blob/2641cccafccace33e20d408489e1832b75757c28/zellij-client/src/stdin_handler.rs#L27
and here
https://github.com/zellij-org/zellij/blob/2641cccafccace33e20d408489e1832b75757c28/zellij-client/src/stdin_handler.rs#L44

test `ctrl+tab` and `ctrl + fxx`, and both got nothing, while it did work when pressing `ctrl+a`, printing `49` . I am a little confused, can't OS read combination like `ctrl+tab` and `ctrl + fxx`, which has nothing to do with termwiz.

**Comment by musjj** [comment]
@imsnif

AFAIK, kitty's [keyboard protocol](https://sw.kovidgoyal.net/kitty/keyboard-protocol) is the best one out there for supporting special key sequences, way better than the fixterms solution (a nice overview [here](https://wezfurlong.org/wezterm/config/key-encoding.html)).

Both Helix and Neovim supports kitty's keyboard protocol:

- https://github.com/helix-editor/helix/pull/4939
- https://github.com/neovim/neovim/issues/14400

Nushell too: https://github.com/nushell/nushell/pull/10540

Terminals I know that implements it (other than kitty): [wezterm](https://wezfurlong.org/wezterm/config/lua/config/enable_kitty_keyboard.html), [foot](https://codeberg.org/dnkl/foot/issues/319) and [rio](https://raphamorim.io/rio/docs/features/kitty-keyboard-protocol). Some more [here](https://sw.kovidgoyal.net/kitty/keyboard-protocol)

It's not the most universal thing out there, but it's a very solid protocol with plenty of implementations in the wild. I think it'd be pretty reasonable to support it.

**Comment by imsnif** [comment]
A week ago, I would have said that I don't think this is true @musjj 

The threshold for me is always either one of:
1. libvte implemented it - meaning all the major linux terminals have it (gnome-terminal, konsole, xfce-terminal, etc.)
2. Alacritty implemented it - the most popular OSS terminal

But yesterday, Alacritty implemented it: https://github.com/alacritty/alacritty/releases/tag/v0.13.0

So yes, I agree. We should implement support for it. I hope to get to it early this year (2024), no promises though.

**Comment by azinsharaf** [comment]
hi @imsnif , I am wondering if there is any plan to add `Ctrl Tab` to the keybinding support. I'd like to use it in Zellij normal mode to toggle the tabs. Any workaround is appreciated.
```
    normal {
        bind "Ctrl Tab" { ToggleTab; }
    }
```
I use wezterm on Windows and tried enabling `enable_kitty_keyboard` but it introduced some other issues with neovim ( i couldn't change the neovim modes)

**Comment by azinsharaf** [comment]
UPDATE: 
I was able to use ctrl tab to toggle the tabs in zellij. I changed some wezterm and neovim configs to avoid any conflicts with the zellij kitty protocol.


```
    normal {
        bind "Ctrl Tab" { ToggleTab; }
    }
```

changes to neovim and wezterm:


https://github.com/azinsharaf/dotfiles/commit/99925a5e91d45dcc0cb7d86f2447e460b3a0a777

**Comment by calebreed601-dot** [comment]
Thank you for the update. It is great to hear you found a workaround for
using Ctrl + Tab by adjusting your WezTerm and Neovim configurations.
